### PR TITLE
vmss: (PREVIEW) support cross zone scaleset

### PR DESCRIPTION
**Pull request by yugangw-msft**
The main work is on default logic adjustments for uniqueness coming with LB/Standard so the create command would still be very simple like: `az vmss create -g RG -n VMSS --image ubuntults --zones 1 2`

There is a breaking change that for single zone vmss now CLI will use `Standard` LB instead of `Basic`. This is fine as zone is still a preview feature and Basic LB will not work well with zonal scalesets, hence it is better to pull the trigger early 

Advanced users can still pre-create LB and PIP with any prefer'd configurations and supply to `vmss create`, e.g.
```
$ az network public-ip create -n ${PIPNAME} -g ${RGNAME} --sku Standard
$ az network lb create -n ${LBNAME} -g ${RGNAME} --public-ip-address ${PIPNAME} --sku Standard
$ az network lb rule create --backend-port 5000 --frontend-port 80 --lb-name ${LBNAME} -n ${LBNAME} --protocol Tcp -g ${RGNAME}
$ az vmss create -n ${VMSSNAME} -g ${RGNAME} --image ${IMAGE} --lb ${LBNAME} --nsg ${NSGNAME}
 ```
### General Guidelines

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by yugangw-msft** [comment]
//cc: @gatneil

**Comment by yugangw-msft** [comment]
@tjprescott, could you please take a look? This update comes from a long discussion with CRP and NRP folks and the template change has been reviewed by Neil.

File: src/command_modules/azure-cli-vm/HISTORY.rst [hunk 1]
```diff
@@ -5,6 +5,7 @@ Release History
 
 2.0.24
 ++++++
+* vmss: (PREVIEW) cross zone support
```
Comment (tjprescott) [inline src/command_modules/azure-cli-vm/HISTORY.rst]: We should call out the breaking behavior here.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py [hunk 2]
```diff
@@ -274,7 +274,7 @@ def load_arguments(self, _):
         c.argument('backend_pool_name', help='Name to use for the backend pool when creating a new load balancer or application gateway.')
         c.argument('backend_port', help='When creating a new load balancer, backend port to open with NAT rules (Defaults to 22 on Linux and 3389 on Windows). When creating an application gateway, the backend port to use for the backend HTTP settings.', type=int)
         c.argument('load_balancer', help='Name to use when creating a new load balancer (default) or referencing an existing one. Can also reference an existing load balancer by ID or specify "" for none.', options_list=['--load-balancer', '--lb'])
-        c.argument('load_balancer_sku', resource_type=ResourceType.MGMT_NETWORK, min_api='2017-08-01', help='SKU when creating a new Load Balancer.', options_list=['--lb-sku'], arg_type=get_enum_type(LoadBalancerSkuName, default='Basic'))
+        c.argument('load_balancer_sku', resource_type=ResourceType.MGMT_NETWORK, min_api='2017-08-01', help='SKU when creating a new Load Balancer.', options_list=['--lb-sku'], arg_type=get_enum_type(LoadBalancerSkuName))
```
Comment (tjprescott) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py]: Is there no longer a default or is it dynamically determined?  If it is dynamic we should explain that in the help.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py [hunk 3]
```diff
@@ -143,8 +143,8 @@ def build_public_ip_resource(cmd, name, location, tags, address_allocation, dns_
         'dependsOn': [],
         'properties': public_ip_properties
     }
-    if zone:
-        public_ip['zones'] = zone
+    if zone and len(zone) == 1:
+        public_ip['zones'] = zone  # PIP itself doesn't support multiple zones. Standard LB will handle it
```
Comment (tjprescott) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py]: Why is this needed?  If zone doesn't have nargs='+' then how would it ever be >1?  What happens if it is >1?  Right now it looks like it would ignore it. If you somehow get that condition and it is invalid, then an error should be thrown.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py [hunk 4]
```diff
@@ -661,6 +661,24 @@ def build_load_balancer_resource(cmd, name, location, tags, backend_pool_name, n
     }
     if sku and cmd.supported_api_version(ResourceType.MGMT_NETWORK, min_api='2017-08-01'):
         lb['sku'] = {'name': sku}
+        # LB rule is the way to enable SNAT so outbound connections are possible
+        if sku.lower() == 'standard':
```
Comment (tjprescott) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py]: We need to re-record (or at least re-run live) the network tests for LB to ensure that this change doesn't inadvertently break them.  It would not be reflected in simply playing back the recordings.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py [hunk 5]
```diff
@@ -1748,6 +1748,10 @@ def _get_public_ip_address_allocation(value, sku):
 
     # Handle load balancer creation
     if load_balancer_type == 'new':
+        # Defaults SKU to 'Standard' for zonal scale set
+        if load_balancer_sku is None:
+            load_balancer_sku = 'Standard' if zones else 'Basic'
```
Comment (tjprescott) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/custom.py]: Ah, the answer to my earlier question.  We need to call this behavior out in the help.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py (hunk 3 above)
Comment (tjprescott) [reply src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py]: Please update the comment to reflect our discussion.

Comment (yugangw-msft) [reply src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py]: Done

File: src/command_modules/azure-cli-vm/HISTORY.rst (hunk 1 above)
Comment (yugangw-msft) [reply src/command_modules/azure-cli-vm/HISTORY.rst]: Done
